subsys/net/lib/nrf_provisioning/src/nrf_provisioning.c

```diff
@@ -177,18 +177,32 @@ static int nrf_provisioning_set(const char *key, size_t len_rd,
 	key_len = settings_name_next(key, &next);
 
 	if (strncmp(key, "interval-sec", key_len) == 0) {
-		len = read_cb(cb_arg, &time_str, sizeof(time_str));
-		if (len <= 0) {
+		if (len_rd >= sizeof(time_str)) {
+			return -ENOMEM;
+		}
+
+		len = read_cb(cb_arg, &time_str, len_rd);
+		if (len < 0) {
 			LOG_ERR("Unable to read the timestamp of next provisioning");
-			memset(&time_str, 0, sizeof(time_str));
+			return len;
 		}
 
 		if (strlen(time_str) == 0) {
```

Comment:
I think you need to null-terminate the string and also tolerate len=0 which means that the key is deleted.

```suggestion
		time_str[len] = 0;

		if (len == 0) {
```

See https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/settings/settings.h#L61-L64